event-sourcing/event-publisher/event-publisher-process/src/main/java/uk/gov/justice/services/eventsourcing/publishing/SequenceNumberUpdater.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.justice.services.eventsourcing.publishing;
+
+import static javax.transaction.Transactional.TxType.MANDATORY;
+
+import uk.gov.justice.subscription.registry.SubscriptionDataSourceProvider;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+public class SequenceNumberUpdater {
+
+    private static final String NEXT_VAL_FROM_SEQUENCE_SQL = "SELECT nextval('event_sequence_number_seq')";
+    private static final String SELECT_PREVIOUS_SEQUENCE_NUMBER_SQL = "SELECT sequence_number FROM event_sequence WHERE next = 0 FOR UPDATE";
+    private static final String UPDATE_NEXT_COLUMN_IN_PREVIOUS_ROW_SQL = "UPDATE event_sequence SET next = ? WHERE next = 0";
+    private static final String INSERT_NEW_ROW_SQL = "INSERT INTO event_sequence (event_id, sequence_number, previous, next) VALUES (?, ?, ?, ?)";
+
+    @Inject
+    SubscriptionDataSourceProvider subscriptionDataSourceProvider;
+
+    @Transactional(MANDATORY)
+    public long update(final UUID eventId) throws SQLException {
+
+        try (final Connection connection = subscriptionDataSourceProvider.getEventStoreDataSource().getConnection()) {
+
+            final long nextValFromPostgresSequence = nextValFromPostgresSequence(connection);
+            final long previous = getPrevious(connection);
+
+            updateNextColumnInPreviousRow(nextValFromPostgresSequence, connection);
+            insertRow(eventId, nextValFromPostgresSequence, previous, connection);
+
+            return nextValFromPostgresSequence;
+        }
+    }
+
+    private long nextValFromPostgresSequence(final Connection connection) throws SQLException {
+
+        try (final PreparedStatement preparedStatement = connection.prepareStatement(NEXT_VAL_FROM_SEQUENCE_SQL)) {
+            try (final ResultSet resultSet = preparedStatement.executeQuery()) {
+                resultSet.next();
+                return resultSet.getLong(1);
+            }
+        }
+    }
+
+    private long getPrevious(final Connection connection) throws SQLException {
+
+        try(final PreparedStatement preparedStatement = connection.prepareStatement(SELECT_PREVIOUS_SEQUENCE_NUMBER_SQL)) {
+            try(final ResultSet resultSet = preparedStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    return resultSet.getLong(1);
+                }
+            }
+        }
+
+        return 0;
```

Comment:
Could make it clear why zero is returned here...maybe a constant would make the intent clear